<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential analysis with CATALYST • CATALYST</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/simplex/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="Differential analysis with CATALYST">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-54095998-3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-54095998-3');
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CATALYST</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Release version">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Documentation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Guides
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/differential_analysis.html">Differential analysis with CATALYST</a>
    </li>
    <li>
      <a href="../articles/preprocessing.html">Preprocessing with CATALYST</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">News</a>
</li>
<li>
  <a href="https://github.com/HelenaLC/CATALYST">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="http://www.imls.uzh.ch/en/research/robinson.html">Robinson Lab</a>
</li>
<li>
  <a href="http://www.bodenmillerlab.org">Bodenmiller Lab</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Differential analysis with CATALYST</h1>
                        <h4 class="author">Helena L Crowell</h4>
            <address class="author_afil">
      Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><a class="author_email" href="mailto:#"></a><a href="mailto:crowellh@student.ethz.ch">crowellh@student.ethz.ch</a>
      </address>
                              <h4 class="author">Mark D Robinson</h4>
            <address class="author_afil">
      Institute for Molecular Life Sciences, University of Zurich, Switzerland<br><h4 class="date">16 May 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/HelenaLC/CATALYST/blob/master/vignettes/differential_analysis.Rmd"><code>vignettes/differential_analysis.Rmd</code></a></small>
      <div class="hidden name"><code>differential_analysis.Rmd</code></div>

    </address>
</div>

    
    
<hr>
<p><span style="color:firebrick"><strong>Most of the pipeline and visualizations presented herein were adapted from <span class="citation">Nowicka et al. (2017)</span>’s <em>“CyTOF workflow: differential discovery in high-throughput high-dimensional cytometry datasets”</em>. For the complete workflow, go <a href="10.12688/f1000research.11622.2">here</a>.</strong></span></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="co"># load required packages</span></a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw">suppressPackageStartupMessages</span>({</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">    <span class="kw">library</span>(CATALYST)</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">    <span class="kw">library</span>(SummarizedExperiment)</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">})</a></code></pre></div>
<div id="example-data" class="section level1">
<h1 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h1>
<ul>
<li>
<code>PBMC_fs</code>:<br>
a <code>flowSet</code> holding PBMCs samples from 4 patients, each containing between 500 and 1000 cells. For each sample, the expression of 10 cell surface and 14 signaling markers was measured before (REF) and upon BCR/FcR-XL stimulation (BCRXL) with B cell receptor/Fc receptor crosslinking for 30’, resulting in a total of 8 samples.<br>
This data set represents a subset of data originating from <span class="citation">Bodenmiller et al. (2012)</span> that was also used in the <a href="https://github.com/nolanlab/citrus">citrus</a> paper <span class="citation">(Bruggner et al. 2014)</span>.</li>
<li>
<code>PBMC_panel</code>:<br>
a data.frame containing each marker’s column name in the FCS file (<code>fcs_colname</code> column), and its targeted protein marker (<code>antigen</code> column).</li>
<li>
<code>PBMC_md</code>:<br>
a data.frame where rows correspond to samples, and columns specify each sample’s <code>file_name</code>, <code>sample_id</code>, <code>condition</code>, and <code>patient_id</code>.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># load example data</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="../reference/data.html">data</a></span>(PBMC_fs, PBMC_panel, PBMC_md)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3">PBMC_fs</a></code></pre></div>
<pre><code>## A flowSet with 8 experiments.
## 
##   column names:
##   CD3(110:114)Dd CD45(In115)Dd pNFkB(Nd142)Dd pp38(Nd144)Dd CD4(Nd145)Dd CD20(Sm147)Dd CD33(Nd148)Dd pStat5(Nd150)Dd CD123(Eu151)Dd pAkt(Sm152)Dd pStat1(Eu153)Dd pSHP2(Sm154)Dd pZap70(Gd156)Dd pStat3(Gd158)Dd CD14(Gd160)Dd pSlp76(Dy164)Dd pBtk(Er166)Dd pPlcg2(Er167)Dd pErk(Er168)Dd pLat(Er170)Dd IgM(Yb171)Dd pS6(Yb172)Dd HLA-DR(Yb174)Dd CD7(Yb176)Dd</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="kw">head</span>(PBMC_panel)</a></code></pre></div>
<pre><code>##      fcs_colname antigen
## 1 CD3(110:114)Dd     CD3
## 2  CD45(In115)Dd    CD45
## 3 pNFkB(Nd142)Dd   pNFkB
## 4  pp38(Nd144)Dd    pp38
## 5   CD4(Nd145)Dd     CD4
## 6  CD20(Sm147)Dd    CD20</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw">head</span>(PBMC_md)</a></code></pre></div>
<pre><code>##                 file_name sample_id condition patient_id
## 1 PBMC_patient1_BCRXL.fcs    BCRXL1     BCRXL   Patient1
## 2   PBMC_patient1_Ref.fcs      Ref1       Ref   Patient1
## 3 PBMC_patient2_BCRXL.fcs    BCRXL2     BCRXL   Patient2
## 4   PBMC_patient2_Ref.fcs      Ref2       Ref   Patient2
## 5 PBMC_patient3_BCRXL.fcs    BCRXL3     BCRXL   Patient3
## 6   PBMC_patient3_Ref.fcs      Ref3       Ref   Patient3</code></pre>
<p>The code snippet below demonstrates how to construct a <code>flowSet</code> from a set of FCS files. However, we also give the option to directly specify the path to a set of FCS files (see next section).</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1"><span class="co"># download exemplary set of FCS files</span></a>
<a class="sourceLine" id="cb8-2" data-line-number="2">url &lt;-<span class="st"> "http://imlspenticton.uzh.ch/robinson_lab/cytofWorkflow"</span></a>
<a class="sourceLine" id="cb8-3" data-line-number="3">fcs_zip &lt;-<span class="st"> "PBMC8_fcs_files.zip"</span></a>
<a class="sourceLine" id="cb8-4" data-line-number="4"><span class="kw">download.file</span>(<span class="kw">paste0</span>(url, <span class="st">"/"</span>, fcs_zip), <span class="dt">destfile =</span> fcs_zip, <span class="dt">mode =</span> <span class="st">"wb"</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5"><span class="kw">unzip</span>(fcs_zip)</a>
<a class="sourceLine" id="cb8-6" data-line-number="6"></a>
<a class="sourceLine" id="cb8-7" data-line-number="7"><span class="co"># read in FCS files as flowSet</span></a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="kw">library</span>(flowCore)</a>
<a class="sourceLine" id="cb8-9" data-line-number="9">fcs_files &lt;-<span class="st"> </span><span class="kw">list.files</span>(<span class="dt">pattern =</span> <span class="st">".fcs$"</span>)</a>
<a class="sourceLine" id="cb8-10" data-line-number="10">fs &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/flowCore/topics/read.flowSet">read.flowSet</a></span>(fcs_files, <span class="dt">transformation =</span> <span class="ot">FALSE</span>, <span class="dt">truncate_max_range =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
</div>
<div id="data-organization-the-daframe-class" class="section level1">
<h1 class="hasAnchor">
<a href="#data-organization-the-daframe-class" class="anchor"></a>Data organization: The <code>daFrame</code> class</h1>
<p>Data used and returned throughout differential analysis are held in objects of the <code>daFrame</code> class. Its constructor requires the following inputs:</p>
<ul>
<li>
<code>x</code>: a <code>flowSet</code> holding the raw measurement data, or a character string that specifies a path to a set of FCS files.</li>
<li>
<code>panel</code>: a 2 column data.frame that contains for each marker of interest i) its column name in the raw input data, and ii) its targeted protein marker.</li>
<li>
<code>md</code>: a data.frame with columns describing the experimental design.</li>
</ul>
<p>Optionally, <code>cols_to_use</code> will specify which columns (channels) to keep from the input data. Here, we keep all measurement parameters (default value <code>cols_to_use = NULL</code>).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1"><span class="co"># construct daFrame</span></a>
<a class="sourceLine" id="cb9-2" data-line-number="2">(daF &lt;-<span class="st"> </span><span class="kw"><a href="../reference/daFrame-class.html">daFrame</a></span>(PBMC_fs, PBMC_panel, PBMC_md))</a></code></pre></div>
<pre><code>## class: daFrame 
## dim: 5428 24 
## metadata(2): experiment_info n_cells
## assays(1): exprs
## rownames: NULL
## rowData names(3): sample_id condition patient_id
## colnames(24): CD3 CD45 ... HLA_DR CD7
## colData names(3): channel_name marker_name marker_class</code></pre>
<p>We provide flexibility in the way the panel and metadata table can be set up. Specifically, column names are allowed to differ from the example above, and multiple factors (patient ID, conditions, batch etc.) can be specified. Arguments <code>panel_cols</code> and <code>md_cols</code> should then be used to specify which columns hold the required information. An example is given below:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># alter panel column names</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">panel2 &lt;-<span class="st"> </span>PBMC_panel</a>
<a class="sourceLine" id="cb11-3" data-line-number="3"><span class="kw">colnames</span>(panel2) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"channel_name"</span>, <span class="st">"marker"</span>)</a>
<a class="sourceLine" id="cb11-4" data-line-number="4"></a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="co"># alter metadata column names &amp; add 2nd condition</span></a>
<a class="sourceLine" id="cb11-6" data-line-number="6">md2 &lt;-<span class="st"> </span>PBMC_md</a>
<a class="sourceLine" id="cb11-7" data-line-number="7"><span class="kw">colnames</span>(md2) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"file"</span>, <span class="st">"sampleID"</span>, <span class="st">"cond1"</span>, <span class="st">"patientID"</span>)</a>
<a class="sourceLine" id="cb11-8" data-line-number="8">md2<span class="op">$</span>cond2 &lt;-<span class="st"> </span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>), <span class="dv">4</span>)</a>
<a class="sourceLine" id="cb11-9" data-line-number="9"></a>
<a class="sourceLine" id="cb11-10" data-line-number="10"><span class="co"># construct daFrame</span></a>
<a class="sourceLine" id="cb11-11" data-line-number="11"><span class="kw"><a href="../reference/daFrame-class.html">daFrame</a></span>(PBMC_fs, panel2, md2, </a>
<a class="sourceLine" id="cb11-12" data-line-number="12">    <span class="dt">panel_cols =</span> <span class="kw">list</span>(<span class="dt">channel =</span> <span class="st">"channel_name"</span>, <span class="dt">antigen =</span> <span class="st">"marker"</span>),</a>
<a class="sourceLine" id="cb11-13" data-line-number="13">    <span class="dt">md_cols =</span> <span class="kw">list</span>(<span class="dt">file =</span> <span class="st">"file"</span>, <span class="dt">id =</span> <span class="st">"sampleID"</span>, </a>
<a class="sourceLine" id="cb11-14" data-line-number="14">        <span class="dt">factors =</span> <span class="kw">c</span>(<span class="st">"cond1"</span>, <span class="st">"cond2"</span>, <span class="st">"patientID"</span>)))</a></code></pre></div>
<p>Note that, independent of the input panel and metadata tables, the constructor will fix the names of mandatory slots for latter data accession (<code>sample_id</code> in the <code>rowData</code>, <code>channel_name</code> and <code>marker_name</code> in the <code>colData</code>). The <code>md</code> table will be stored under <code>experiment_info</code> inside the <code>metadata</code>.</p>
</div>
<div id="diagnostic-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#diagnostic-plots" class="anchor"></a>Diagnostic plots</h1>
<div id="plotcounts-number-of-cells-measured-per-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#plotcounts-number-of-cells-measured-per-sample" class="anchor"></a><code>plotCounts</code>: Number of cells measured per sample</h2>
<p>The number of cells measured per sample may be plotted with <code>plotCounts</code>, or directly accessed via <code>n_cells()</code>. This plot should be used as a guide together with other readouts to identify samples where not enough cells were assayed.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw">n_cells</span>(daF)</a></code></pre></div>
<pre><code>## BCRXL1   Ref1 BCRXL2   Ref2 BCRXL3   Ref3 BCRXL4   Ref4 
##    528    881    665    438    563    660    934    759</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1"><span class="kw"><a href="../reference/plotCounts.html">plotCounts</a></span>(daF, <span class="dt">color_by =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-6-1.png" width="576"></p>
</div>
<div id="plotmds-multi-dimensional-scaling-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#plotmds-multi-dimensional-scaling-plot" class="anchor"></a><code>plotMDS</code>: Multi-dimensional scaling plot</h2>
<p>A multi-dimensional scaling (MDS) plot on median expresion values may be rendered with <code>plotMDS</code>. Such a plot will give a sense of similarities between samples in an unsupervised way and of key difference in expression before conducting any formal testing. In our example, we can see a clear separation between reference (REF) and stimulation condition (BCRXL).</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="../reference/plotMDS.html">plotMDS</a></span>(daF, <span class="dt">color_by =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<pre><code>## Warning: package 'bindrcpp' was built under R version 3.4.4</code></pre>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-7-1.png" width="480"></p>
</div>
<div id="plotexprheatmap-heatmap-of-scaled-median-marker-expressions" class="section level2">
<h2 class="hasAnchor">
<a href="#plotexprheatmap-heatmap-of-scaled-median-marker-expressions" class="anchor"></a><code>plotExprHeatmap</code>: Heatmap of (scaled) median marker expressions</h2>
<p><code>plotExprHeatmap</code> will show a heatmap on median marker intensities with hierarchically clustered columns (samples) and rows (markers). This plot should give an idea of which markers will drive sample clustering, and how similiar samples are in their expression profile.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="../reference/plotExprHeatmap.html">plotExprHeatmap</a></span>(daF, <span class="dt">color_by =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-8-1.png" width="960"></p>
</div>
</div>
<div id="clustering" class="section level1">
<h1 class="hasAnchor">
<a href="#clustering" class="anchor"></a>Clustering</h1>
<div id="cluster-flowsom-clustering-consensusclusterplus-metaclustering" class="section level2">
<h2 class="hasAnchor">
<a href="#cluster-flowsom-clustering-consensusclusterplus-metaclustering" class="anchor"></a><code>cluster</code>: <em>FlowSOM</em> clustering &amp; <em>ConsensusClusterPlus</em> metaclustering</h2>
<p><em><a href="http://bioconductor.org/packages/CATALYST">CATALYST</a></em> provides a simple wrapper to perform high resolution <code>FlowSOM</code> clustering and lower resolution <code>ConsensusClusterPlus</code> metaclustering. By default, the data will be initially clustered into <code>xdim = 10</code> x <code>ydim = 10</code> = 100 groups. Secondly, the function will metacluster populations into 2 through <code>maxK</code> (default 20) clusters. To make analyses reproducible, the random seed may be set via <code>seed</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="co"># specify markers to use for clustering</span></a>
<a class="sourceLine" id="cb18-2" data-line-number="2">lineage_markers &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"CD3"</span>, <span class="st">"CD45"</span>, <span class="st">"CD4"</span>, <span class="st">"CD20"</span>, </a>
<a class="sourceLine" id="cb18-3" data-line-number="3">    <span class="st">"CD33"</span>, <span class="st">"CD123"</span>, <span class="st">"CD14"</span>, <span class="st">"IgM"</span>, <span class="st">"HLA_DR"</span>, <span class="st">"CD7"</span>)</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">daF &lt;-<span class="st"> </span><span class="kw"><a href="../reference/cluster.html">cluster</a></span>(daF, <span class="dt">cols_to_use =</span> lineage_markers, </a>
<a class="sourceLine" id="cb18-5" data-line-number="5">    <span class="dt">xdim =</span> <span class="dv">10</span>, <span class="dt">ydim =</span> <span class="dv">10</span>, <span class="dt">maxK =</span> <span class="dv">20</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>, <span class="dt">seed =</span> <span class="dv">1</span>)       </a></code></pre></div>
<p>Let K = <code>xdim</code> x <code>ydim</code> be the number of <em><a href="http://bioconductor.org/packages/FlowSOM">FlowSOM</a></em> clusters. <code>cluster</code> will add information to the following slots of the input <code>daFrame</code>:</p>
<ul>
<li>
<code>rowData</code>:
<ul>
<li>
<code>cluster_id</code>: cluster ID as inferred by <em><a href="http://bioconductor.org/packages/FlowSOM">FlowSOM</a></em>. One of 1, …, K.</li>
</ul>
</li>
<li>
<code>colData</code>:
<ul>
<li>
<code>marker_class</code>: factor <code>"type"</code> or <code>"state"</code>. Specifyies whether a marker has been used for clustering or not, respectively.</li>
</ul>
</li>
<li>
<code>metadata</code>:
<ul>
<li>
<code>SOM_codes</code>: a table with dimensions K x (# type markers). Contains the SOM codes.</li>
<li>
<code>cluster_codes</code>: a table with dimensions K x (<code>maxK</code> + 1). Contains the cluster codes for all metaclusterings.</li>
<li>
<code>delta_area</code>: a <code>ggplot</code> object (see below for details).</li>
</ul>
</li>
</ul>
</div>
<div id="delta-area-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#delta-area-plot" class="anchor"></a>Delta area plot</h2>
<p>The delta area represents the amount of extra cluster stability gained when clustering into k groups as compared to k-1 groups. It can be expected that high stability of clusters can be reached when clustering into the number of groups that best fits the data. The “natural” number of clusters present in the data should thus corresponds to the value of k where there is no longer a considerable increase in stability (pleateau onset). For more details, the user can refer to the original description of the consensus clustering method <span class="citation">(Monti et al. 2003)</span>.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" data-line-number="1"><span class="co"># access &amp; render delta area plot</span></a>
<a class="sourceLine" id="cb19-2" data-line-number="2"><span class="kw">metadata</span>(daF)<span class="op">$</span>delta_area</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-10-1.png" width="480"></p>
</div>
<div id="mergeclusters-manual-cluster-merging" class="section level2">
<h2 class="hasAnchor">
<a href="#mergeclusters-manual-cluster-merging" class="anchor"></a><code>mergeClusters</code>: Manual cluster merging</h2>
<p>Provided with a 2 column data.frame containing <code>old_cluster</code> and <code>new_cluster</code> IDs, <code>mergeClusters</code> allows for manual cluster merging of any clustering available within the input <code>daFrame</code> (i.e. the <code>xdim</code> x <code>ydim</code> <em><a href="http://bioconductor.org/packages/FlowSOM">FlowSOM</a></em> clusters, and any of the 2-<code>maxK</code> <em><a href="http://bioconductor.org/packages/ConsensusClusterPlus">ConsensusClusterPlus</a></em> metaclusters). For latter accession (visualization, differential testing), the function will assign a unique ID (specified with <code>id</code>) to each merging, and add a column to the <code>cluster_codes</code> inside the <code>metadata</code> slot of the input <code>daFrame</code>.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" data-line-number="1"><span class="kw"><a href="../reference/data.html">data</a></span>(merging_table)</a>
<a class="sourceLine" id="cb20-2" data-line-number="2"><span class="kw">head</span>(merging_table)</a></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   old_cluster new_cluster 
##         &lt;dbl&gt; &lt;chr&gt;       
## 1           1 B-cells IgM+
## 2           2 surface-    
## 3           3 NK cells    
## 4           4 CD8 T-cells 
## 5           5 B-cells IgM-
## 6           6 monocytes</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" data-line-number="1">daF &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mergeClusters.html">mergeClusters</a></span>(daF, <span class="dt">table =</span> merging_table, <span class="dt">id =</span> <span class="st">"merging1"</span>)</a>
<a class="sourceLine" id="cb22-2" data-line-number="2"><span class="kw">head</span>(<span class="kw">cluster_codes</span>(daF))</a></code></pre></div>
<pre><code>##   100 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20     merging1
## 1   1 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1 B-cells IgM+
## 2   2 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1 B-cells IgM+
## 3   3 1 1 1 1 1 1 1 1  1  1  1  1  1  1  1  1  1  1  1 B-cells IgM+
## 4   4 1 1 1 1 1 2 2 2  2  2  2  2  2  2  2  2  2  2  2     surface-
## 5   5 1 1 1 1 1 2 2 2  2  2  2  2  2  2  2  2  2  2  2     surface-
## 6   6 1 1 1 1 1 2 2 2  2  2  2  2  2  3  3  3  3  3  3     NK cells</code></pre>
</div>
<div id="plotclusterheatmap-heatmap-of-metaclustering-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotclusterheatmap-heatmap-of-metaclustering-results" class="anchor"></a><code>plotClusterHeatmap</code>: Heatmap of (meta)clustering results</h2>
<p>Clusterings and metaclusters maybe be viewing with the <code>plotClusterHeatmap</code>. In its 1st panel, the function will display median (arcsinh-transformed and <em>optionally</em> scaled) cell-type marker expressions (across all samples). Depending on argument <code>hm2</code>, the 2nd panel will vary as follows:</p>
<ul>
<li>
<code>"abundances"</code>: cluster frequencies by sample;</li>
<li>
<code>"state_markers"</code>: median cell state marker expressions across clusters (analogous to the left-hand side heatmap);</li>
<li>a character string/vector corresponding to one/multiple marker(s): median marker expressions by sample.</li>
</ul>
<p>Argument <code>scale</code> (default <code>TRUE</code>) specifies whether scaled values should be plotted. These correspond to arcsinh-transformed expression values scaled between 0 and 1 using low (1%) and high (99%) percentiles as boundaries. Note that, in any case, hierarchical clustering is performed on the unscaled data.<br>
While it is only used here for visualization, this additional transformation of the arcsinh-transformed data can sometimes give a better representation of relative differences in marker expression between cell populations.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" data-line-number="1"><span class="co"># median pS6 expression by sample as 2nd heatmap</span></a>
<a class="sourceLine" id="cb24-2" data-line-number="2"><span class="kw"><a href="../reference/plotClusterHeatmap.html">plotClusterHeatmap</a></span>(daF, <span class="dt">hm2 =</span> <span class="st">"pS6"</span>, <span class="dt">k=</span><span class="dv">12</span>, <span class="dt">m =</span> <span class="dv">6</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" data-line-number="1"><span class="co"># population frequencies by sample as 2nd heatmap</span></a>
<a class="sourceLine" id="cb25-2" data-line-number="2"><span class="kw"><a href="../reference/plotClusterHeatmap.html">plotClusterHeatmap</a></span>(daF, <span class="dt">hm2 =</span> <span class="st">"abundances"</span>, </a>
<a class="sourceLine" id="cb25-3" data-line-number="3">    <span class="dt">draw_freqs =</span> <span class="ot">TRUE</span>, <span class="dt">cluster_anno =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-13-1.png" width="960"></p>
</div>
<div id="plotabundances-relative-population-abundances" class="section level2">
<h2 class="hasAnchor">
<a href="#plotabundances-relative-population-abundances" class="anchor"></a><code>plotAbundances</code>: Relative population abundances</h2>
<p>Relative population abundances for any clustering of interest can be plotted with <code>plotAbundances</code>. Argument <code>by</code> will specify whether to plot proportions for each sample or cluster.<br>
If <code>by = "sample_id"</code>, the function displays each sample’s cell type composition, and the size of a given stripe reflects the proportion of the corresponding cell type the given sample. Argument <code>group</code> then specifies the facetting. If <code>by = "cluster_id"</code>, argument <code>group</code> then specifies the grouping and color coding.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" data-line-number="1"><span class="kw"><a href="../reference/plotAbundances.html">plotAbundances</a></span>(daF, <span class="dt">k =</span> <span class="dv">12</span>, <span class="dt">by =</span> <span class="st">"sample_id"</span>, <span class="dt">group =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-14-1.png" width="576"></p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" data-line-number="1"><span class="kw"><a href="../reference/plotAbundances.html">plotAbundances</a></span>(daF, <span class="dt">k =</span> <span class="st">"merging1"</span>, <span class="dt">by =</span> <span class="st">"cluster_id"</span>, <span class="dt">group =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-14-2.png" width="576"></p>
</div>
</div>
<div id="dimensionality-reduction-with-tsne" class="section level1">
<h1 class="hasAnchor">
<a href="#dimensionality-reduction-with-tsne" class="anchor"></a>Dimensionality reduction with <code>tSNE</code>
</h1>
<p>Dimensionality reduction through t-SNE may be run via <code>tSNE</code>. Here, the subset of markers to use is specified with <code>cols_to_use</code>, and the number of cells to dowsample to per sample is given by <code>n</code>. Again, the random seed may be set via <code>seed</code> for reproducibility.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" data-line-number="1">daF &lt;-<span class="st"> </span><span class="kw"><a href="../reference/tSNE.html">tSNE</a></span>(daF, <span class="dt">cols_to_use =</span> lineage_markers, </a>
<a class="sourceLine" id="cb28-2" data-line-number="2">    <span class="dt">n =</span> <span class="dv">200</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>, <span class="dt">seed =</span> <span class="dv">42</span>)</a></code></pre></div>
<div id="plotsne-t-sne-visualization" class="section level2">
<h2 class="hasAnchor">
<a href="#plotsne-t-sne-visualization" class="anchor"></a><code>plotSNE</code>: t-SNE visualization</h2>
<p><code>plotSNE</code> is a flexbile wrapper to plot the t-SNE colored by clustering or marker expression. Furthermore, the data may be split by condition, sample ID, or any other factor available in the <code>daFrame</code>’s <code>rowData</code>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" data-line-number="1"><span class="co"># color by pS6 expression and split by condition</span></a>
<a class="sourceLine" id="cb29-2" data-line-number="2"><span class="kw"><a href="../reference/plotSNE.html">plotSNE</a></span>(daF, <span class="dt">color_by =</span> <span class="st">"pS6"</span>, <span class="dt">facet =</span> <span class="st">"condition"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-16-1.png" width="672"></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" data-line-number="1"><span class="co"># color by 8 metaclusters &amp; split by sample ID</span></a>
<a class="sourceLine" id="cb30-2" data-line-number="2"><span class="kw"><a href="../reference/plotSNE.html">plotSNE</a></span>(daF, <span class="dt">color_by =</span> <span class="dv">8</span>, <span class="dt">facet =</span> <span class="st">"sample_id"</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-17-1.png" width="768"></p>
</div>
</div>
<div id="differental-testing-with-diffcyt" class="section level1">
<h1 class="hasAnchor">
<a href="#differental-testing-with-diffcyt" class="anchor"></a>Differental testing with <em><a href="http://bioconductor.org/packages/diffcyt">diffcyt</a></em>
</h1>
<p><em><a href="http://bioconductor.org/packages/CATALYST">CATALYST</a></em> has been designed to be compatible with the <em><a href="http://bioconductor.org/packages/diffcyt">diffcyt</a></em> package <span class="citation">(Weber 2018)</span>, which implements statistical methods for differential discovery in high-dimensional cytometry (including flow cytometry, mass cytometry or CyTOF, and oligonucleotide-tagged cytometry) using high-resolution clustering and moderated tests. The input to the <em><a href="http://bioconductor.org/packages/diffcyt">diffcyt</a></em> pipeline can either be raw data, or a <code>daFrame</code> object. We give an exmaple of the latter below.<br>
Please refer to the <em><a href="http://bioconductor.org/packages/diffcyt">diffcyt</a></em> <a href="https://bioconductor.org/packages/3.7/bioc/vignettes/diffcyt/inst/doc/diffcyt_workflow.html">vignette</a> and R documentation (<code>??diffcyt</code>) for more detailed information.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" data-line-number="1"><span class="kw">library</span>(diffcyt)</a>
<a class="sourceLine" id="cb31-2" data-line-number="2"><span class="co"># create design &amp; constrast matrix</span></a>
<a class="sourceLine" id="cb31-3" data-line-number="3">design &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/diffcyt/topics/createDesignMatrix">createDesignMatrix</a></span>(PBMC_md, <span class="dt">cols_design=</span><span class="dv">3</span>)</a>
<a class="sourceLine" id="cb31-4" data-line-number="4">contrast &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/diffcyt/topics/createContrast">createContrast</a></span>(<span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb31-5" data-line-number="5"><span class="co"># test for</span></a>
<a class="sourceLine" id="cb31-6" data-line-number="6"><span class="co"># - differential abundance (DA) of clusters</span></a>
<a class="sourceLine" id="cb31-7" data-line-number="7"><span class="co"># - differential states (DS) within clusters</span></a>
<a class="sourceLine" id="cb31-8" data-line-number="8">res_DA &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/diffcyt/topics/diffcyt">diffcyt</a></span>(daF, <span class="dt">design =</span> design, <span class="dt">contrast =</span> contrast, </a>
<a class="sourceLine" id="cb31-9" data-line-number="9">    <span class="dt">analysis_type =</span> <span class="st">"DA"</span>, <span class="dt">method_DA =</span> <span class="st">"diffcyt-DA-edgeR"</span>)</a>
<a class="sourceLine" id="cb31-10" data-line-number="10">res_DS &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/diffcyt/topics/diffcyt">diffcyt</a></span>(daF, <span class="dt">design =</span> design, <span class="dt">contrast =</span> contrast, </a>
<a class="sourceLine" id="cb31-11" data-line-number="11">    <span class="dt">analysis_type =</span> <span class="st">"DS"</span>, <span class="dt">method_DS =</span> <span class="st">"diffcyt-DS-limma"</span>, <span class="dt">plot =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<div id="plotdiffheatmap-heatmap-of-differential-testing-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotdiffheatmap-heatmap-of-differential-testing-results" class="anchor"></a><code>plotDiffHeatmap</code>: Heatmap of differential testing results</h2>
<p>Differential testing results returned by <em><a href="http://bioconductor.org/packages/diffcyt">diffcyt</a></em> can be displayed with the <code>plotDiffHeatmap</code> function.</p>
<p>For differential abundance (DA) tests, <code>plotDiffHeatmap</code> will display</p>
<ul>
<li>median (arcsinh-transformed) cell-type marker expressions (across all samples), and</li>
<li>cluster abundances by samples</li>
</ul>
<p>For differential state (DS) tests, <code>plotDiffHeatmap</code> will display</p>
<ul>
<li>median (arcsinh-transformed) cell-type marker expressions (across all samples)</li>
<li>median (arcsinh-transformed) cell-state marker expressions by sample</li>
</ul>
<p>Clusters (DA) and cluster-marker combinations (DS), respectively, will be marked as <em>significant</em> if their adjusted p-value falls below the threshold value specified with <code>th</code> (default 0.1), and will be ordered by significance if <code>order = TRUE</code> (the default). The number of top findings to display can be specified with <code>top_n</code> (default 20).</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" data-line-number="1"><span class="kw"><a href="../reference/plotDiffHeatmap.html">plotDiffHeatmap</a></span>(daF, res_DA, <span class="dt">th =</span> <span class="fl">1e-3</span>, <span class="dt">top_n =</span> <span class="dv">25</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-19-1.png" width="960"></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" data-line-number="1"><span class="kw"><a href="../reference/plotDiffHeatmap.html">plotDiffHeatmap</a></span>(daF, res_DS, <span class="dt">top_n =</span> <span class="dv">10</span>)</a></code></pre></div>
<p><img src="differential_analysis_files/figure-html/unnamed-chunk-19-2.png" width="960"></p>
</div>
</div>
<div id="session-information" class="section level1">
<h1 class="hasAnchor">
<a href="#session-information" class="anchor"></a>Session Information</h1>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" data-line-number="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 3.4.1 (2017-06-30)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.4
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] diffcyt_0.99.15            bindrcpp_0.2.2            
##  [3] SummarizedExperiment_1.8.1 DelayedArray_0.4.1        
##  [5] matrixStats_0.53.1         Biobase_2.38.0            
##  [7] GenomicRanges_1.30.3       GenomeInfoDb_1.14.0       
##  [9] IRanges_2.12.0             S4Vectors_0.16.0          
## [11] BiocGenerics_0.24.0        CATALYST_1.5.2            
## [13] BiocStyle_2.6.1           
## 
## loaded via a namespace (and not attached):
##   [1] minqa_1.2.4                 TH.data_1.0-8              
##   [3] Rtsne_0.13                  colorspace_1.3-2           
##   [5] rjson_0.2.18                rio_0.5.10                 
##   [7] rprojroot_1.3-2             circlize_0.4.3             
##   [9] corpcor_1.6.9               XVector_0.18.0             
##  [11] GlobalOptions_0.0.13        fs_1.2.2                   
##  [13] rstudioapi_0.7              roxygen2_6.0.1             
##  [15] ggrepel_0.8.0               DT_0.4                     
##  [17] mvtnorm_1.0-7               xml2_1.2.0                 
##  [19] codetools_0.2-15            splines_3.4.1              
##  [21] robustbase_0.93-0           knitr_1.20                 
##  [23] jsonlite_1.5                nloptr_1.0.4               
##  [25] cluster_2.0.7-1             shinydashboard_0.7.0       
##  [27] graph_1.56.0                shiny_1.0.5                
##  [29] rrcov_1.4-3                 compiler_3.4.1             
##  [31] httr_1.3.1                  drc_3.0-1                  
##  [33] backports_1.1.2             assertthat_0.2.0           
##  [35] Matrix_1.2-14               lazyeval_0.2.1             
##  [37] cli_1.0.0                   limma_3.34.9               
##  [39] later_0.7.2                 htmltools_0.3.6            
##  [41] tools_3.4.1                 igraph_1.2.1               
##  [43] gtable_0.2.0                glue_1.2.0                 
##  [45] GenomeInfoDbData_1.0.0      reshape2_1.4.3             
##  [47] dplyr_0.7.4                 Rcpp_0.12.16               
##  [49] carData_3.0-1               cellranger_1.1.0           
##  [51] pkgdown_1.0.0               nlme_3.1-137               
##  [53] xfun_0.1                    stringr_1.3.1              
##  [55] lme4_1.1-17                 openxlsx_4.0.17            
##  [57] mime_0.5                    gtools_3.5.0               
##  [59] XML_3.98-1.11               edgeR_3.20.9               
##  [61] DEoptimR_1.0-8              zlibbioc_1.24.0            
##  [63] MASS_7.3-50                 zoo_1.8-1                  
##  [65] scales_0.5.0                shinyBS_0.61               
##  [67] promises_1.0.1              sandwich_2.4-0             
##  [69] RColorBrewer_1.1-2          ComplexHeatmap_1.17.1      
##  [71] yaml_2.1.19                 curl_3.2                   
##  [73] memoise_1.1.0               gridExtra_2.3              
##  [75] ggplot2_2.2.1               stringi_1.2.2              
##  [77] desc_1.2.0                  pcaPP_1.9-73               
##  [79] plotrix_3.7-1               flowCore_1.44.2            
##  [81] shape_1.4.4                 rlang_0.2.0                
##  [83] pkgconfig_2.0.1             commonmark_1.5             
##  [85] bitops_1.0-6                evaluate_0.10.1            
##  [87] lattice_0.20-35             purrr_0.2.4                
##  [89] bindr_0.1.1                 labeling_0.3               
##  [91] htmlwidgets_1.2             plyr_1.8.4                 
##  [93] magrittr_1.5                bookdown_0.7               
##  [95] R6_2.2.2                    nnls_1.4                   
##  [97] multcomp_1.4-8              pillar_1.2.2               
##  [99] haven_1.1.1                 foreign_0.8-70             
## [101] survival_2.42-3             abind_1.4-5                
## [103] RCurl_1.95-4.10             FlowSOM_1.10.0             
## [105] tibble_1.4.2                tsne_0.1-3                 
## [107] crayon_1.3.4                car_3.0-0                  
## [109] utf8_1.1.3                  plotly_4.7.1               
## [111] rmarkdown_1.9               GetoptLong_0.1.6           
## [113] locfit_1.5-9.1              grid_3.4.1                 
## [115] readxl_1.1.0                data.table_1.10.4-3        
## [117] ConsensusClusterPlus_1.42.0 forcats_0.3.0              
## [119] digest_0.6.15               xtable_1.8-2               
## [121] httpuv_1.4.3                tidyr_0.8.0                
## [123] munsell_0.4.3               viridisLite_0.3.0          
## [125] shinyjs_1.0</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Bodenmiller12">
<p>Bodenmiller, Bernd, Eli R Zunder, Rachel Finck, Tiffany J Chen, Erica S Savig, Robert V Bruggner, Erin F Simonds, et al. 2012. “Multiplexed Mass Cytometry Profiling of Cellular States Perturbed by Small-Molecule Regulators.” <em>Nature Biotechnology</em> 30 (9):858–67. <a href="https://doi.org/10.1038/nbt.2317" class="uri">https://doi.org/10.1038/nbt.2317</a>.</p>
</div>
<div id="ref-Bruggner14">
<p>Bruggner, Robert V, Bernd Bodenmiller, David L Dill, Robert J Tibshirani, and Garry P Nolan. 2014. “Automated Identification of Stratifying Signatures in Cellular Subpopulations.” <em>Proceedings of the National Academy of Sciences of the United States of America</em> 111 (26). National Academy of Sciences:E2770–E2777. <a href="https://doi.org/10.1073/pnas.1408792111" class="uri">https://doi.org/10.1073/pnas.1408792111</a>.</p>
</div>
<div id="ref-Monti03">
<p>Monti, Stefano, Pablo Tamayo, Jill Mesirov, and Todd Golub. 2003. “Consensus Clustering: A Resampling-Based Method for Class Discovery and Visualization of Gene Expression Microarray Data.” <em>Machine Learning</em> 52 (1):91–118. <a href="https://doi.org/10.1023/A:1023949509487" class="uri">https://doi.org/10.1023/A:1023949509487</a>.</p>
</div>
<div id="ref-Nowicka17">
<p>Nowicka, M, C Krieg, LM Weber, FJ Hartmann, S Guglietta, B Becher, MP Levesque, and MD Robinson. 2017. “CyTOF Workflow: Differential Discovery in High-Throughput High-Dimensional Cytometry Datasets [Version 2; Referees: 2 Approved].” <em>F1000Research</em> 6 (748). <a href="https://doi.org/10.12688/f1000research.11622.2" class="uri">https://doi.org/10.12688/f1000research.11622.2</a>.</p>
</div>
<div id="ref-diffcyt">
<p>Weber, LM. 2018. “Diffcyt: Differential Discovery in High-Dimensional Cytometry via High-Resolution Clustering.” <em>R Package Version 1.0.0</em>. <a href="https://github.com/lmweber/diffcyt" class="uri">https://github.com/lmweber/diffcyt</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#example-data">Example data</a></li>
      <li><a href="#data-organization-the-daframe-class">Data organization: The <code>daFrame</code> class</a></li>
      <li>
<a href="#diagnostic-plots">Diagnostic plots</a><ul class="nav nav-pills nav-stacked">
<li><a href="#plotcounts-number-of-cells-measured-per-sample"><code>plotCounts</code>: Number of cells measured per sample</a></li>
      <li><a href="#plotmds-multi-dimensional-scaling-plot"><code>plotMDS</code>: Multi-dimensional scaling plot</a></li>
      <li><a href="#plotexprheatmap-heatmap-of-scaled-median-marker-expressions"><code>plotExprHeatmap</code>: Heatmap of (scaled) median marker expressions</a></li>
      </ul>
</li>
      <li>
<a href="#clustering">Clustering</a><ul class="nav nav-pills nav-stacked">
<li><a href="#cluster-flowsom-clustering-consensusclusterplus-metaclustering"><code>cluster</code>: <em>FlowSOM</em> clustering &amp; <em>ConsensusClusterPlus</em> metaclustering</a></li>
      <li><a href="#delta-area-plot">Delta area plot</a></li>
      <li><a href="#mergeclusters-manual-cluster-merging"><code>mergeClusters</code>: Manual cluster merging</a></li>
      <li><a href="#plotclusterheatmap-heatmap-of-metaclustering-results"><code>plotClusterHeatmap</code>: Heatmap of (meta)clustering results</a></li>
      <li><a href="#plotabundances-relative-population-abundances"><code>plotAbundances</code>: Relative population abundances</a></li>
      </ul>
</li>
      <li>
<a href="#dimensionality-reduction-with-tsne">Dimensionality reduction with <code>tSNE</code></a><ul class="nav nav-pills nav-stacked">
<li><a href="#plotsne-t-sne-visualization"><code>plotSNE</code>: t-SNE visualization</a></li>
      </ul>
</li>
      <li>
<a href="#differental-testing-with-diffcyt">Differental testing with <em><a href="http://bioconductor-org/packages/diffcyt">diffcyt</a></em></a><ul class="nav nav-pills nav-stacked">
<li><a href="#plotdiffheatmap-heatmap-of-differential-testing-results"><code>plotDiffHeatmap</code>: Heatmap of differential testing results</a></li>
      </ul>
</li>
      <li><a href="#session-information">Session Information</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Helena L. Crowell, Vito R.T. Zanotelli, Stéphane Chevrier, Mark D. Robinson, Bernd Bodenmiller.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
